### DISCUSSION TITLE
Row wrapper in VGrid

### DISCUSSION THREAD

**Discussion by manelio** [inokawa/virtua#512] [Q&A] [answered]
In my project, I use a kind of large tables. The concept is the same as VGrid but with a wrapper for each row. Something like:

```
{rowVirtualizer.map(row => (
  <div className='this-is-a-row'>
    {columnVirtualizer.map(column => ...)}
  </div>
))}
```

The reason is simply to make highlighting and row selection easier.

I want to migrate to virtua from @tanstack/react-virtual, where a grid is two nested virtualizers (in my case, a window virtualizer for rows and regular virtualizer for columns).

Is there an example of this use case?

**Comment by inokawa** [comment]
2 direction virtualization with VGrid is still experimental. It may not work as expected for now.

**Reply by inokawa** [reply, marked as answer]
Shipped stabilized `VGrid` in `0.52.0`.
https://inokawa.github.io/virtua/?path=/story/advanced-datatable--data-table
